### PR TITLE
Test: Compute Res Auxiliar Value

### PR DESCRIPTION
**Pull request by jkktom**
Made chage to Add function in memory_value.go
also in vm.go changed addOperand case.
the main issue resolved. (felt-to-address addition)

the reason was felt memory is created but not discarded (to Nil), (by GoLang behavior) 
So just make it Memory address and let add function handle with new memory space
Please review and comment.
Thanks.

### CODE REVIEW

File: pkg/vm/vm.go [hunk 1]
```diff
@@ -345,7 +345,7 @@ func (vm *VirtualMachine) computeRes(
 	case AddOperands:
 		op0 := op0Cell.Read()
 		op1 := op1Cell.Read()
-		return mem.EmptyMemoryValueAs(op0.IsAddress()).Add(op0, op1)
+		return mem.EmptyMemoryValueAsAddress().Add(op0, op1)
```
Comment (rodrigo-pino) [inline pkg/vm/vm.go:348]:
```suggestion
return mem.EmptyMemoryValue(op0.IsAddress() || op1.IsAddress()).Add(op0, op1)
```

Comment (jkktom) [reply pkg/vm/vm.go]: Applied as discussed via meeting

File: pkg/vm/memory/memory_value.go [hunk 2]
```diff
@@ -212,25 +211,26 @@ func (mv *MemoryValue) Equal(other *MemoryValue) bool {
...
-	if lhs.IsAddress() {
-		if !rhs.IsFelt() {
-			return nil, errors.New("rhs is not a felt")
-		}
-		mv.address, err = mv.address.Add(lhs.address, rhs.felt)
-	} else {
-		if rhs.IsAddress() {
+
+	// If both lhs and rhs are felts, perform a simple addition
+	if lhs.IsFelt() {
```
Comment (jkktom) [inline pkg/vm/memory/memory_value.go:218]: Better intuitive / readability with IsFelt() method.

File: pkg/vm/vm.go [hunk 3]
```diff
@@ -345,7 +345,7 @@ func (vm *VirtualMachine) computeRes(
 	case AddOperands:
 		op0 := op0Cell.Read()
 		op1 := op1Cell.Read()
-		return mem.EmptyMemoryValueAs(op0.IsAddress()).Add(op0, op1)
+		return mem.EmptyMemoryValueAs(op0.IsAddress() || op1.IsAddress()).Add(op0, op1)
```
Comment (jkktom) [inline pkg/vm/vm.go:348]: Review meeting ; suggested by Rodrigo and resolves prior problems (Felt + Addr addition error)

**Review by rodrigo-pino — APPROVED** [review]

**Pull request merged**